### PR TITLE
Improve Scope analysis and add a tutorial doc

### PR DESCRIPTION
**Pull request by jimmylai**
## Summary
Changes:
1. add `Scope.get_assignments_for()` to make accessing Assignments easier while traversing CST tree.
With existing API, user needs to parse the name string from CSTNode and then use `scope[name]`.
This new API is similar to `get_qualified_names_for()` but returns assignments instead when given a CSTNode.

2. implement Scope.__iter__ for accessing all assignments in current scope.

3. improve document of Scope: after vs before
image

4. add scope tutorial to demonstrate some advanced use cases.
  - Automatically Remove Unused Import

https://2776-200896124-gh.circle-artifacts.com/0/doc/scope_tutorial.html

## Test Plan
unit tests.

### CODE REVIEW

File: libcst/metadata/scope_provider.py [hunk 1]
```diff
@@ -284,6 +292,14 @@ def f(self) -> "c":
                 )
         return results
 
+    def get_assignments_for(self, node: cst.CSTNode) -> Collection[BaseAssignment]:
+        """ Get all assignments in current scope given a :class:`~libcst.CSTNode`. """
+        full_name = _QualifiedNameUtil.get_full_name_for(node)
+        if full_name is None:
+            return set()
+        parts = full_name.split(".")
+        return self[parts[0]]
```
Comment (DragonMinded) [inline libcst/metadata/scope_provider.py]: I'm not sure on the implications of claiming that "a.b" assignments are equivalent to "a" assignments. Do we have this documented anywhere?

File: docs/source/scope_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    "    print(scope)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "We're interested in those unused names and we traverse all import assignments in each scope to build a lookup table ``import_names_to_remove`` to record the mapping of import node and their unused names. There exist other types of assignment like :class:`~libcst.FunctionDef` or :class:`~libcst.ClassDef` but we don't care them here."
```
Comment (DragonMinded) [inline docs/source/scope_tutorial.ipynb]: Missing about in "care them here".

File: libcst/metadata/scope_provider.py (hunk 1 above)
Comment (jimmylai) [reply libcst/metadata/scope_provider.py]: We decided to use Assignments class to replace this function and we'll clarify the current scope analysis only handles local variable name.

**Comment by jimmylai** [comment]
Comments are addressed and added to PR #94.
We decided to replace  get_assignments_for with Assignments/Accesses.
Also improved the tutorial.

**Pull request closed without merging**